test/test_lagrangian_particle_tracking.jl

```diff
@@ -255,6 +263,43 @@ function run_simple_particle_tracking_tests(arch, timestepper; vertically_stretc
     @test all(w .≈ 0)
     @test all(s .≈ √2)
 
+    # Test active tracers
+    ts = arch_array(arch, ones(P))
+    particles = StructArray{TestParticle}((xs, ys, zs, us, vs, ws, ss, ts))
+
+    velocities = VelocityFields(grid)
+    u, v, w = velocities
+    speed = Field(√(u*u + v*v + w*w))
+    tracked_fields = merge(velocities, (; s=speed))
+    active_properties = ((property=:t, tracer=:t), )
+
+    lagrangian_particles = LagrangianParticles(particles; tracked_fields, parameters=(; active_properties))
+    @test lagrangian_particles isa LagrangianParticles
+
+    model = NonhydrostaticModel(; grid, timestepper, velocities, particles=lagrangian_particles, tracers=:t)
+
+    set!(model, u=1, v=1)
+
+    sim = Simulation(model, Δt=1e-2, stop_iteration=1)
+
+    jld2_filepath = "test_particles.jld2"
+    sim.output_writers[:particles_jld2] =
+        JLD2OutputWriter(model, (; particles=model.particles),
+                         filename=jld2_filepath, schedule=IterationInterval(1))
+
+    nc_filepath = "test_particles.nc"
+    sim.output_writers[:particles_nc] =
+        NetCDFOutputWriter(model, model.particles, filename=nc_filepath, schedule=IterationInterval(1))
+
+    sim.output_writers[:checkpointer] = Checkpointer(model, schedule=IterationInterval(1),
+                                                     dir = ".", prefix = "particles_checkpoint")
+
+    model = sim.model
+    run!(sim)
+
+    ni, nj, nk, d = Oceananigans.LagrangianParticleTracking.getmask(0.0, 0.0, 0.5, grid, (Center(), Center(), Center()))
+    vols = [Oceananigans.Operators.volume(i, nj[idx], nk[idx], grid, Center(), Center(), Center()) for (idx, i) in enumerate(ni)]
```

Comment:
Corresponds to changes [here](https://github.com/CliMA/Oceananigans.jl/pull/2724/files#r964377399)

```suggestion
    nodes, normfactor = getmask(0.0, 0.0, 0.5, grid, (Center(), Center(), Center()))
    vols = [Oceananigans.Operators.volume(getnode(topo[1](), i, grid.Nx), getnode(topo[2](), j, grid.Ny), getnode(topo[3](), k, grid.Nz), grid, Center(), Center(), Center()) for (i, j, k, d) in nodes]
```

Reply:
Same general comments as above; writing this on more lines will help with readability